core/meta/src/TStatusBitsChecker.cxx

```diff
@@ -0,0 +1,272 @@
+Error in <TStatusBitsChecker>:    Bit  13 used in TStreamerElement as kDoNotDelete
+Error in <TStatusBitsChecker>:    Bit  13 used in TObject as kInvalidObject
+  ~~~ {.cpp}
+
+*/
+
+#include "TStatusBitsChecker.h"
+
+#include "TClass.h"
+#include "TBaseClass.h"
+#include "TClassTable.h"
+#include "TEnum.h"
+#include "TEnumConstant.h"
+#include "TError.h"
+
+#include <cmath>
+
+namespace ROOT {
+namespace Detail {
+
+struct TStatusBitsChecker::Registry::Info {
+   Info() = default;
+   Info(const Info&) = default;
+   Info(Info &&) = default;
+
+   Info(TClass &o, std::string &&n, bool intentionalDup) :
+      fOwner(&o), fConstantName(n), fIntentionalDup(intentionalDup) {}
+
+   ~Info() = default;
+
+   TClass *fOwner;
+   std::string fConstantName;
+   bool fIntentionalDup = false;
+};
+
+/// Default constructor. Implemented in source file to allow hiding of the Info struct.
+TStatusBitsChecker::Registry::Registry() = default;
+
+/// Default destructor. Implemented in source file to allow hiding of the Info struct.
+TStatusBitsChecker::Registry::~Registry() = default;
+
+/// Figure out which bit the constant has been set from/to.
+/// Return 255 if the constant is not an integer or out of range.
+UChar_t TStatusBitsChecker::ConvertToBit(Long64_t constant, TClass &classRef, const char *constantName) {
+
+   if (constant <= 0) {
+      Error("TStatusBitsChecker::ConvertBit",
+            "In %s the value of %s is %lld which was not produced by BIT macro.",
+            classRef.GetName(), constantName, constant);
+      return 255;
+   }
+
+   int backshift;
+   double fraction = std::frexp(constant, &backshift);
+   // frexp doc is:
+   //    if no errors occur,
+   //    returns the value x in the range (-1;-0.5], [0.5; 1)
+   //    and stores an integer value in *exp such that x×2^(*exp) = arg
+   --backshift; // frexp is such that BIT(0) == 1 == 0.5*2^(*exp) with *exp == 1
+
+   if (backshift < 0 || std::abs(0.5-fraction) > 0.00001) {
+      Error("TStatusBitsChecker::ConvertBit",
+            "In %s the value of %s is %lld which was not produced by BIT macro.",
+            classRef.GetName(), constantName, constant);
+      return 255;
+   }
+
+   if (backshift > 24) {
+      Error("TStatusBitsChecker::ConvertBit",
+            "In %s the value of %s is %lld (>23) which is ignored by SetBit.",
+            classRef.GetName(), constantName, constant);
+
+      if (backshift > 255) // really we could snip it sooner.
+         backshift = 255;
+   }
+
+   return backshift;
+}
+
+void TStatusBitsChecker::Registry::RegisterBits(TClass &classRef /* = false */)
+{
+   TEnum *eStatusBits = (TEnum*)classRef.GetListOfEnums()->FindObject("EStatusBits");
+   TEnum *exceptionBits = (TEnum*)classRef.GetListOfEnums()->FindObject("EStatusBitsDupExceptions");
+
+   if (eStatusBits) {
+
+      for(auto c : *eStatusBits->GetConstants()) {
+         TEnumConstant *constant = dynamic_cast<TEnumConstant*>(c);
```

Comment:
Coverity will be unhappy with you :-) What if the dyn_cast fails?

Reply:
It can't :) the 'original' version of the code was 
```
TEnumConstant *constant = (TEnumConstant*)(c);
```